src/ajax/xhr.js

```diff
@@ -88,12 +88,15 @@ jQuery.ajaxTransport( function( options ) {
 									xhrSuccessStatus[ xhr.status ] || xhr.status,
 									xhr.statusText,
 
-									// Support: IE9
-									// Accessing binary-data responseText throws an exception
-									// (#11426)
-									typeof xhr.responseText === "string" ? {
-										text: xhr.responseText
-									} : undefined,
+									// XHR2 responseText throws an exception on binary data,
+									// return it raw and let a converter or caller handle it
+									// https://xhr.spec.whatwg.org/#the-responsetext-attribute
+									// (trac-11426, gh-2498)
+									xhr.responseType === "arraybuffer" ||
```

Comment:
Alternatively, we could look for the cases we wanted to force into the `text` path, but that's a little more code since XHR1 doesn't have `.responseType` and we actually want to handle JSON or XML conversions ourselves via `text`.